WATOLINK/bci-gaming
# Game Music & Endless Background

## ClickUp ticket link
[Game Music](https://sharing.clickup.com/14173911/v/4-60744162-2/t/h/8685qf32b/6e8ef21b1bbc311)
[Endless Background](https://app.clickup.com/t/8685qf3bn)

## Implementation description
- Game Music: 
   -  Plays music on load. The object will not be destroyed when transitioning into different scenes (i.e the background music will continue playing across different scenes)
   - It is a prefab that can just be dragged into the scene.

- Endless Background:
   - Background tracks the camera position. If the camera is moving too far to the right, the background moves to the right (same as left)  
   - It is a prefab that can just be dragged into the scene.

## Steps to test
1. Go under Assets >> Prefabs and look for the "BackgroundMusicGameManager" and "Background"
2. Drag the Prefabs into the scene.
3. For the BackgroundMusicGameManager, see if the music plays. For the Background, see if it is creating an "Endless" background effect.

## What should reviewers focus on?

Background Music: When you navigate into a different scene the music should still play. When you navigate back to the menu, there should only be one BackgroundMusicGameManager playing music. 

Background: Making sure the endless effect is smooth

## Checklist
